libraries/ui-library/src/components/six-rating/six-rating.tsx

```diff
@@ -0,0 +1,203 @@
+
+  /** If this property is set to true and an error message is provided by `errorText`, the error message is displayed.  */
+  @Prop({ reflect: true }) invalid = false;
+
+  /** Maximum number of stars. Default is 5 */
+  @Prop() max: number = 5;
+
+  /** Size of the stars */
+  @Prop({ reflect: true }) size: 'xSmall' | 'small' | 'medium' | 'large' | 'xLarge' | 'xxLarge' | 'xxxLarge' = 'medium';
+
+  /** If its readonly */
+  @Prop() readonly: boolean = false;
+
+  /** Emitted when the control loses focus. */
+  @Event({ eventName: 'six-rating-blur' }) sixBlur!: EventEmitter<number>;
+
+  /** Emitted when the control's checked state changes. */
+  @Event({ eventName: 'six-rating-change' }) sixChange!: EventEmitter<number>;
+
+  /** Emitted when the control gains focus. */
+  @Event({ eventName: 'six-rating-focus' }) sixFocus!: EventEmitter<EmptyPayload>;
+
+  // --
+
+  @State() hasFocus = false;
+  @State() hasErrorTextSlot = false;
+  @State() hoveredValue = this.value;
+
+  private static id: number = 0;
+
+  private ratingId = `rating-${++SixRating.id}`;
+  private labelId = `rating-label-${SixRating.id}`;
+  private errorTextId = `input-error-text-${SixRating.id}`;
+
+  @Watch('value')
+  handleValueChange() {
+    this.sixChange.emit(this.value);
+  }
+
+  connectedCallback() {
+    this.el.shadowRoot?.addEventListener('slotchange', this.handleSlotChange);
+  }
+
+  disconnectedCallback() {
+    this.el.shadowRoot?.removeEventListener('slotchange', this.handleSlotChange);
+  }
+
+  private handleSlotChange = () => {
+    this.hasErrorTextSlot = hasSlot(this.el, 'error-text');
+  };
+
+  private handleBlur = () => {
+    this.hasFocus = false;
+    this.sixBlur.emit(this.value);
+  };
+
+  private handleFocus = () => {
+    this.hasFocus = true;
+    this.sixFocus.emit();
+  };
+
+  private handleKeyDown = (event: KeyboardEvent) => {
+    if (event.key === 'ArrowLeft') {
+      event.preventDefault();
+      if (this.hoveredValue > 0) {
+        this.hoveredValue = this.hoveredValue - 1;
+        this.value = this.hoveredValue;
+      }
+    }
+
+    if (event.key === 'ArrowRight') {
+      event.preventDefault();
+      if (this.hoveredValue < this.max) {
+        this.hoveredValue = this.hoveredValue + 1;
+      }
+    }
+  };
+
+  private renderStars() {
+    return Array.from({ length: this.max }, (_, i) => {
+      const handleEvent = (event: Event) => {
+        if (this.readonly || this.disabled) {
+          return;
+        }
+        switch (event.type) {
+          case 'mouseover':
+            this.hoveredValue = i + 1;
+            break;
+          case 'mouseout':
+            this.hoveredValue = this.value;
+            break;
+          case 'click':
+            this.value = this.hoveredValue;
+            break;
+        }
+      };
+
+      return (
+        <six-icon
+          class={{
+            'six-rating__star': true,
+            'six-rating__star--filled': i < this.hoveredValue || i < this.value,
```

Comment:
```suggestion
            'six-rating__star--filled': i < this.hoveredValue,
```
In my opinion, it should always behave the same when hovering. The stars should react to the position of the mouse, even if the hovered values is less than the current one. What do you thing?